test/test_grid.py

```diff
@@ -214,3 +211,42 @@ def test_compute_face_areas_fesom(self):
         grid_2_ds = xr.open_dataset(fesom_grid_small)
         grid_2 = ux.Grid(grid_2_ds)
         grid_2.compute_face_areas()
+
+
+class TestPopulateCoordinates(TestCase):
+
+    def test_populate_cartesian_xyz_coord(self):
+        verts_lonlat = np.array([[0, -80], [95, 10], [290, 77]])
+        vgrid = ux.Grid(verts_lonlat)
+        vgrid._populate_cartesian_xyz_coord()
+        num_nodes = vgrid.ds.Mesh2_node_x.size
+        for i in range(0, num_nodes):
+            [x, y, z] = [
+                vgrid.ds["Mesh2_node_cart_x"].values[i],
+                vgrid.ds["Mesh2_node_cart_y"].values[i],
+                vgrid.ds["Mesh2_node_cart_z"].values[i]
+            ]
+            [x_cal, y_cal, z_cal] = ux.helpers._convert_node_lonlat_rad_to_xyz(
+                np.deg2rad(verts_lonlat[i].tolist()))
+            nt.assert_almost_equal(x, x_cal, decimal=12)
+            nt.assert_almost_equal(y, y_cal, decimal=12)
+            nt.assert_almost_equal(z, z_cal, decimal=12)
+
+    def test_populate_lonlat_coord(self):
+        """Create a uxarray grid from vertices ."""
+        verts = np.array([[0.57735027, -5.77350269e-01, -0.57735027],
+                          [0.57735027, 5.77350269e-01, -0.57735027],
+                          [-0.57735027, 5.77350269e-01, -0.57735027]])
+        vgrid = ux.Grid(verts)
+        vgrid.ds.Mesh2_node_x.attrs["units"] = "meters"
+        vgrid.ds.Mesh2_node_y.attrs["units"] = "meters"
+        vgrid._populate_lonlat_coord()
+        num_nodes = vgrid.ds.Mesh2_node_x.size
+        for i in range(0, num_nodes):
+            lon = vgrid.ds["Mesh2_node_x"].values[i]
+            lat = vgrid.ds["Mesh2_node_y"].values[i]
+            nodes = list(verts[i])
+            [lon_cal,
+             lat_cal] = ux.helpers._convert_node_xyz_to_lonlat_rad(nodes)
+            nt.assert_almost_equal(np.deg2rad(lon), lon_cal, decimal=12)
+            nt.assert_almost_equal(np.deg2rad(lat), lat_cal, decimal=12)
```

Comment:
I would re-think both of these test cases. When you call `ux.Grid._populate_....`, you are then running the same algorithm in the for loop, essentially testing the function against itself. 

To test the correctness of your functions, I suggest doing the follwing:
1. Create two toy datasets. One should be the pre-converted values $D_{test}$ and the other should be the expected values after running the algorithm/function $D_{gold}$. Note that $D_{gold}$ should be manually computed from $D_{test}$ using something other than the algorithm you are testing
2. Run `ux.Grid._populate_....` on $D_{test}$ to obtain $D_{out}$. 
3. Compare $D_{out}$ to $D_{gold}$

Reply:
Thank you for your comments here.

I just wondering what other algorithms I should use to generate the expected results? I have also been bugged by this problem for a while. So far I can only find one conversion method as implemented in the `ux.helpers._convert_node_xyz_to_lonlat_rad` or vice versa. `ux.helpers._convert_node_xyz_to_lonlat_rad` have passed their own test cases and being merged into the main already. 

I wonder if you have an any better idea than that? Thanks

Reply:
For the expected results, I suggest using some known algorithm that you can compute the old-fashioned way using pen and paper and apply it to a small dataset for testing. It doesn't have to be anything large, just enough to test the function that you wrote.